pegs_common_utils.R

```diff
@@ -262,63 +262,374 @@ prepare_ukb_phenotype <- function(ukb_data, phenotype) {
+    d <- pegs_data[c(exclusion_cols)] %>% 
+      mutate(sum = rowSums(.,na.rm = T)) %>% dplyr::select(sum,everything())
+    epr_number <- pegs_data$epr_number
+    ex_df <- cbind(epr_number,d) %>% dplyr::select(epr_number,sum)
+    he_vars <- pegs_data %>% dplyr::select(epr_number,he_i059_iron_anemia)
+    pheno_data <- left_join(he_vars,ex_df, by = 'epr_number') %>% 
+      mutate(Flag = ifelse(sum == 0,0,1)) %>% 
+      mutate(Iron_Def_Anemia = case_when(he_i059_iron_anemia == 1 ~ 1,
+                                         he_i059_iron_anemia == 0 & Flag == 0 ~ 0,
+                                         TRUE ~ NA_real_)) %>%  
+      dplyr::select(epr_number,Iron_Def_Anemia)
+    return(pheno_data)
+    
+  }
+  
+  .ovarian_cysts_phenotype <- function(pegs_data){ 
+    f_he <- pegs_data %>% dplyr::filter(`_he_gender_`==1)
+    f_he[] <- lapply(f_he, function(x) as.numeric(as.character(x))) 
+    cancer_cols <- names(pegs_data)[grep('cancer',names(pegs_data),ignore.case = T)]
+    cancer_ex <- c('age|sis|bro|dad|mom')
+    cancer_cols <- cancer_cols[-grep(cancer_ex,cancer_cols,ignore.case = T)]
+    exclusion_cols <- c('he_m091_uterine_tumors','he_m089_endometriosis','he_m090_uterine_polyps')
+    f_ex <- f_he[c(exclusion_cols)] %>% 
+      mutate(sum = rowSums(.,na.rm = T)) %>% dplyr::select(sum,everything())
+    epr_number <- f_he$epr_number
+    ex_df <- cbind(epr_number,f_ex) %>% dplyr::select(epr_number,sum)
+    he_vars <- f_he %>% dplyr::select(epr_number,he_m092_ovarian_cysts)
+    pheno_data <- left_join(he_vars,ex_df, by = 'epr_number') %>% 
+      mutate(Flag = ifelse(sum == 0,0,1)) %>% 
+      mutate(Ovarian_Cysts = case_when(he_m092_ovarian_cysts == 1 ~ 1,
+                                       he_m092_ovarian_cysts == 0 & Flag == 0 ~ 0,
+                                       TRUE ~ NA_real_)) %>%  
+      dplyr::select(epr_number,Ovarian_Cysts)
+    return(pheno_data)
+    
+    
+  }
+  
+  .asthma_phenotype <- function(pegs_data){
+    exclusion_cols <- c('he_d025_copd','he_d026_ipf','he_d027_tb_PARQ',"he_d028_cough_breathlessness","he_d029_chest_wheeze")
+    ex_df <- pegs_data[c(exclusion_cols)] 
+    ex_df[] <- lapply(ex_df, function(x) as.numeric(as.character(x))) 
+    d <- ex_df %>% 
+      as.data.frame(.) %>% 
+      mutate(sum = rowSums(.,na.rm = T)) %>% dplyr::select(sum,everything())
+    epr_number <- pegs_data$epr_number
+    
+    ex_df <- cbind(epr_number,d) %>% dplyr::select(epr_number,sum)
+    he_vars <- pegs_data %>% dplyr::select(epr_number,he_d030_asthma_PARQ)
+    pheno_data <- left_join(he_vars,ex_df, by = 'epr_number') %>% 
+      mutate(Flag = ifelse(sum == 0,0,1)) %>% 
+      mutate(Asthma = case_when(he_d030_asthma_PARQ == 1 ~ 1,
+                                he_d030_asthma_PARQ == 0 & Flag == 0 ~ 0,
+                                TRUE ~ NA_real_)) %>% 
+      dplyr::select(epr_number,Asthma)
+    return(pheno_data)
+    
+    
+  }
+  
+  .t2d_phenotype <- function(pegs_data){
```

Comment:
I think we should keep the same criteria as we had before for type2 vs type 1 here, even though that is different than the other phenotypes - we can discuss with Alison.